tests/src/test/scala/com/nvidia/spark/rapids/GpuSemaphoreSuite.scala

```diff
@@ -68,13 +69,4 @@ class GpuSemaphoreSuite extends AnyFunSuite
     GpuSemaphore.releaseIfNecessary(context)
     GpuSemaphore.releaseIfNecessary(context)
   }
-
-  test("Completion listener registered on first acquire") {
```

Comment:
It looks like this test would pass even after this change, as the semaphore registering with ScalableTaskCompletion would cause it to register a new top-level listener callback with the mock context since it hasn't seen a listener callback for that before.

On a related note, could write some simple sanity tests for ScalableTaskCompletionContext to verify that it handles custom task contexts properly (i.e.: not accidentally registering the callback on the wrong context).

Reply:
Will look at adding some unit tests on Monday.